### ISSUE TITLE
Conditional oneliner does not detect lack of &&.

### ISSUE THREAD

**Issue by bert-willekens** [koalaman/shellcheck#2091]
#### For bugs
- Rule Id (if any, e.g. SC1000): None I know.
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
[[ -n "${OPTI}" ]] && OPTI="-iops ${OPTI}"
[[ -n "${OPTM}" ]] OPTM="-bandwidth ${OPTM}"
```

#### Here's what shellcheck currently says:
```
$ shellcheck myscript
No issues detected!
```
#### Here's what I wanted or expected to see:
Bash 4.2 output when running this is:
```
-bash: syntax error near unexpected token `OPTM="-bandwidth ${OPTM}"`
```

I would have liked to receive a warning that I maybe forgot the && (or ||).

**Comment by koalaman** [comment]
Tokens after test expressions and compound commands were skipped, causing a lack of warnings about any such issues. This has now been fixed, and you will get better errors for this and several other previously ignored cases. Thanks!
